### PR TITLE
Fix offset guessing on linux kernel 5.8

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

The previous kprobe function `tcp_get_info` was not being triggered, but rather `tcp_get_info.part.0` was. This moves the kprobe up a level to `tcp_getsockopt` which should be more stable.

### Motivation

The system-probe failed to start on Ubuntu 20.10 which runs linux kernel 5.8. Thanks @ISauve for the bug report.

Note: Ubuntu 20.10 is scheduled to release before the 7.24 agent release, so this should land in 7.23.

### Describe your test plan

All automated eBPF tests pass on Agent VM and Ubuntu 20.10. Further testing can be done by attempting to start `system-probe` since offset guessing happens upon startup.

Please test on CentOS 7.6 and Ubuntu Xenial/16.04 (kernel 4.4)

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

File: pkg/ebpf/c/offset-guess.c [hunk 1]
```diff
@@ -146,15 +147,21 @@ int kprobe__ip_make_skb(struct pt_regs* ctx) {
 }
 
 /* Used exclusively for offset guessing */
-SEC("kprobe/tcp_get_info")
-int kprobe__tcp_get_info(struct pt_regs* ctx) {
+SEC("kprobe/tcp_getsockopt")
+int kprobe__tcp_getsockopt(struct pt_regs* ctx) {
+    int level = (int)PT_REGS_PARM2(ctx);
```
Comment (leeavital) [inline pkg/ebpf/c/offset-guess.c:152]: what does the `level` option control? I couldn't easily tell from looking at the source

**Review by leeavital — APPROVED** [review]
One question.

We should mention testing this older kernels in the QA instructions in case  the tcp_getsockopt isn't stable.

**Review by p-lambert — APPROVED** [review]

**Comment by brycekahle** [comment]
> We should mention testing this older kernels in the QA instructions in case the tcp_getsockopt isn't stable.

@leeavital I looked at kernels back to 3.10 and the arguments were stable.

File: pkg/ebpf/c/offset-guess.c (hunk 1 above)
Comment (brycekahle) [reply pkg/ebpf/c/offset-guess.c:152]: It is the `level` argument from the `getsockopt` system call. https://linux.die.net/man/2/getsockopt

The filters I'm using match what we call `getsockopt` with:
https://github.com/DataDog/datadog-agent/blob/fa0db67f8ed9b4328b453c7157020be4223a9bcf/pkg/ebpf/offsetguess.go#L672

**Pull request merged**